File: src/server/src/model/entities/post.ts
import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, ManyToOne, CreateDateColumn } from "typeorm"
import { Show } from "./show"
import { User } from "./User"

@Entity() 
export class Post extends BaseEntity {
    @PrimaryGeneratedColumn()
    id!: number

    @Column()
    content!: string
    
    @CreateDateColumn()
    createdAt!: Date; 

    @ManyToOne(() => Show)
    show!: Show

    @ManyToOne(() => User, (user) => user.posts, { onDelete: "CASCADE"})
    user!: User
}

File: src/client/lib/features/feed/models/post.dart
class Post {
  final int id;
  final String content;
  final String timestamp;
  final String user;
  final String show;

  Post({
    required this.id,
    required this.content,
    required this.timestamp,
    required this.user,
    required this.show,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'content': content,
      'timestamp': timestamp,
      'user': user,
      'show': show,
    };
  }

  factory Post.fromMap(Map<String, dynamic> map) {
    return Post(
      id: map['id'] ?? 0,
      content: map['content'] ?? '',
      timestamp: map['timestamp'] ?? '',
      user: map['user'] ?? '',
      show: map['show'] ?? '',
    );
  }
}
